src/transformers/data/data_collator.py

```diff
@@ -691,11 +691,15 @@ class DataCollatorForLanguageModeling(DataCollatorMixin):
             tokens and the value to predict for the masked token.
         mlm_probability (`float`, *optional*, defaults to 0.15):
             The probability with which to (randomly) mask tokens in the input, when `mlm` is set to `True`.
+        mask_replace_prob (`float`, *optional*, defaults to 0.8):
+            The probability with which masked tokens are replaced by the tokenizer's mask token (e.g., `[MASK]`).
+            Defaults to 0.8, meaning 80% of the masked tokens will be replaced with `[MASK]`.
+        random_replace_prob (`float`, *optional*, defaults to 0.1):
+            The probability with which masked tokens are replaced by random tokens from the tokenizer's vocabulary.
+            Defaults to 0.1, meaning 10% of the masked tokens will be replaced with random tokens. The remaining
+            masked tokens (1 - mask_replace_prob - random_replace_prob) are left unchanged.
```

Comment:
Clarify here that both of these options only apply when `mlm` is `True`, just in case people are confused!

Reply:
Thanks for the feedback! 
I updated the docstring and added the following tests:
1. **`test_probability_sum_error`**: Ensures an error is raised if `mask_replace_prob + random_replace_prob` is not within `[0, 1]`.  
2. **`test_all_mask_replacement`**: Verifies functionality when `mask_replace_prob=1`, ensuring all tokens are either the original token or `[MASK]`.